src/components/fields/SchemaField.js

```diff
@@ -296,7 +304,35 @@ function SchemaFieldRender(props) {
     uiSchema,
   };
 
-  return <FieldTemplate {...fieldProps}>{field}</FieldTemplate>;
+  const _AnyOfField = registry.fields.AnyOfField;
+
+  return (
+    <FieldTemplate {...fieldProps}>
+      {field}
+
+      {/*
+        If the schema `anyOf` can be rendered as a select control, don't
+        render the `anyOf` selection and let `StringField` component handle
+        rendering
+      */}
+      {schema.anyOf && !isSelect(schema) && (
```

Comment:
I think this logic should move to `getFieldComponent`.

Reply:
Could you elaborate on this? Should `getFieldComponent` return the `AnyOfField` component?

Reply:
Hmm. I see your concern -- the `AnyOfField` as it is right now isn't really the field, but rather something that is used alongside a field. I guess this is workable but it seems bad to me because it splits the logic about what components are used and when. Is it maybe possible to make `AnyOfField` a wrapper around another field (i.e. renders itself and then renders `props.children`)? Then we could have `getFieldComponent` return the `AnyOfField` component.

Reply:
I don't know if that would work either, as `AnyOfField` could render multiple different field components. In `AnyOfField`, rendering is delegated back to `SchemaField` which in turn calls `getFieldComponent`, so `getFieldComponent` still ultimately controls how a schema entry is rendered.
I could put the `AnyOfField` inside `getFieldComponent`, but we then have a situation where we have to add more wrapper markup (because of supporting react 15 we don't have access to `React.Fragment`) to contain the two elements and all of the properties for the `AnyOfField` component need to be passed to `getFieldComponent` so that the `AnyOfField` can be rendered correctly. This seems like a lot of additional complexity for little benefit.

Reply:
OK, you've convinced me. Thanks for thinking this through.